src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -120,6 +120,20 @@ public boolean containsEntryWithId(String id) {
         return internalIDs.contains(id);
     }
 
+    /**
+     * Returns whether an entry with the given bibtex key exists.
+     */
+    public synchronized boolean containsEntryWithKey(String key) {
+        int keyHash = key.hashCode(); // key hash for better performance
```

Comment:
Well, I think it is safe to change getEntryByKey to return the first entry with the key (instead of the last one). And if you at changing this method, then replace the comparison of hashcodes with real equals (like in getEntriesByKey)